### DISCUSSION TITLE
moving the ruler and the compass

### DISCUSSION THREAD

**Discussion by kacago22** [xournalpp/xournalpp#4589] [Ideas]
I see that the compass and the ruler can be moved with two fingers. 

Unfortunately  some drawing tablets (Huion for example) don't respond to touch and I don't see a way of moving them with my active pen. 

I like the ruler in Autodesk SketchBook and you could use it as a model on how to improve the compass and the ruler from xournal++ to be moveable by the active pen. 

PS. I also like how simple the ruler in SketchBook is (with the exception that I cannot see the length of a line while I draw it) and have the feeling that the compass and ruler in xournal++ are way too complicated.

**Comment by rolandlo** [comment]
Touchpad gestures still need to be implemented for the geometry tools. Currently touchpad gestures are only supported for zooming.

**Reply by kacago22** [reply]
Touchpad gestures are really nice, it is unfortunate they are not available on my Huion tablet. I am suggesting the added functionality (like the ruler in SketchBook) on top of the existing one of course.

**Reply by rolandlo** [reply]
Ok, I see. Maybe for the time being you could either use the keyboard (r/R for rotation, s/S for scaling, arrow keys for movement) for moving the geometry tools or bind some tablet buttons to those keys.

**Reply by kacago22** [reply]
Thanks! I did not know about the shortcuts, they are great. As I said - I'm using xournal++ for 20+ hours a week to teach math and other than the occasional freezing I'm very satisfied with it.  These improvement ideas are my way of trying to help the developers., but no hurry, I'm already quite satisfied with it as it is. 

PS. Is there a place where I can see and maybe modify (all) shortcuts?

**Reply by rolandlo** [reply]
You can modify some shortcuts (like those used in the menus) in the glade files (main.glade in particular), but various shortcuts (also for the geometry tools) are hardcoded in different source files, not in any central place. See also https://github.com/xournalpp/xournalpp/issues/919.
